detoks/src/main/java/nl/tudelft/trustchain/detoks/Profile.kt

```diff
@@ -1,31 +1,70 @@
 package nl.tudelft.trustchain.detoks
 
-import android.util.Log
+import android.content.Context
+import com.frostwire.jlibtorrent.TorrentInfo
 import nl.tudelft.trustchain.detoks.gossiper.NetworkSizeGossiper
 
-/**
- * Basic structure for a profile entry
- */
+
 class ProfileEntry(
-    var watchTime: Long = 0, // Average watch time
-    val firstSeen: Long = System.currentTimeMillis()
-)
+    val firstSeen:  Long = System.currentTimeMillis(),
+    var watched:    Boolean = false,    // True if the video was watched
+    var watchTime:  Long = 0,           // Average watch time
+    var duration:   Long = 0,           // Video duration
+    var uploadDate: Long = 0,           // This is the torrent creation date
+    var hopCount:   Int  = 0,           // Amount of other nodes visited
+    var timesSeen:  Int  = 1,           // Count of times we received it
+    var likes:      Int  = 0,           // TODO: Dependent on other team
+) {
+
+}
 
 class Profile(
     val profiles: HashMap<String, ProfileEntry> = HashMap()
 ) {
+
+    fun addProfile(key: String) {
+        if(!profiles.contains(key)) profiles[key] = ProfileEntry()
+        profiles[key] = ProfileEntry()
+    }
+
+    fun addProfile(key: String, info: TorrentInfo) {
+        if(!profiles.contains(key)) profiles[key] = ProfileEntry()
+        profiles[key] = ProfileEntry(uploadDate = (info.creationDate()))
```

Comment:
This can just be a uploadDate assignment. Would remove the entire method since updateEntryUploadDate does the exact same thing.

Reply:
Fixed